drivers/SmartThings/matter-switch/src/switch_handlers/attribute_handlers.lua

```diff
@@ -267,45 +275,91 @@ end
 
 -- [[ ELECTRICAL ENERGY MEASUREMENT CLUSTER ATTRIBUTES ]] --
 
-function AttributeHandlers.cumul_energy_imported_handler(driver, device, ib, response)
-  if ib.data.elements.energy then
-    local watt_hour_value = ib.data.elements.energy.value / fields.CONVERSION_CONST_MILLIWATT_TO_WATT
-    device:set_field(fields.TOTAL_IMPORTED_ENERGY, watt_hour_value, {persist = true})
-    if ib.endpoint_id ~= 0 then
-      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.energyMeter.energy({ value = watt_hour_value, unit = "Wh" }))
+function AttributeHandlers.energy_imported_factory(is_periodic_report)
+  return function(driver, device, ib, response)
+    local state_device = switch_utils.find_child(device, ib.endpoint_id) or device
+    local energy_meter_latest_state = state_device:get_latest_state(
+      "main", capabilities.energyMeter.ID, capabilities.energyMeter.energy.NAME, 0 -- 0 as the default if state is nil
+    )
+    if version.api < 11 then
+      clusters.ElectricalEnergyMeasurement.types.EnergyMeasurementStruct:augment_type(ib.data)
+    end
+    if ib.data.elements.energy then
+      local energy_imported_wh = ib.data.elements.energy.value / fields.CONVERSION_CONST_MILLIWATT_TO_WATT
+      if is_periodic_report then
+        -- handle this report only if cumulative reports are not supported
+        if device:get_field(fields.CUMULATIVE_REPORTS_SUPPORTED) then return end
+        energy_imported_wh = energy_imported_wh + energy_meter_latest_state
+      end
+      device:emit_event_for_endpoint(ib.endpoint_id, capabilities.energyMeter.energy({ value = energy_imported_wh, unit = "Wh" }))
+      local energy_delta_wh = energy_imported_wh - energy_meter_latest_state
+      switch_utils.increment_field(device, fields.TOTAL_IMPORTED_ENERGY, energy_delta_wh, true)
```

Comment:
```suggestion
      device:set_field(fields.TOTAL_IMPORTED_ENERGY, energy_delta_wh, {persist = true})
```
Couldn't this just be this?

Reply:
no, the energy delta from a single report is not equivalent to the total energy

Reply:
I meant to use `energy_imported_wh`

Reply:
As in 
```
      device:set_field(fields.TOTAL_IMPORTED_ENERGY, energy_imported_wh, {persist = true})
```

Reply:
Could you clarify what you mean? It looks like `increment_field` is doing the same thing, just with an extra step

Reply:
increment field is not using the `energy_imported_wh` field, it is taking its own field and incrementing the delta. The delta can come from multiple different endpoints, each of which may have a different total value. These are being summed in this field.

Reply:
Multiple endpoints would not be handled by the same `energyMeter` capability, that is not feasible based on the current profiling logic. Each supported energy cluster would map only to one `energyMeter` capability, mapped across different child devices. So though it's a little confusing where the source of truth is, even using the latest state, we are still sourcing everything directly from specific endpoint event emission.